prov/psm2/src/psmx2_init.c

```diff
@@ -70,9 +70,30 @@ int	 psmx2_tag_layout_locked = 0;
 
 static void psmx2_init_env(void)
 {
+	char *ompi_job_key;
+	psm2_uuid_t uuid = {};
+	unsigned long long *u = (unsigned long long *)uuid;
+
 	if (getenv("OMPI_COMM_WORLD_RANK") || getenv("PMI_RANK") || getenv("PMIX_RANK"))
 		psmx2_env.name_server = 0;
 
+	/*
+	 * OpenMPI 4.x passes the job key via the environment variable, but the
+	 * format is different. Perform format conversion and use it as the
+	 * default uuid. This will be overridden if FI_PSM2_UUID is set, and
+	 * both have lower priority than the auth_key passed via ep_attr.
+	 */
+	ompi_job_key = getenv("OMPI_MCA_orte_precondition_transports");
```

Comment:
It's safe to use this value for Open MPI 4.x, but note that this name is definitely disappearing in Open MPI v5.x.  You'll need to check with @rhc54 and/or others to see what you should key off for Open MPI >= v5.x.

Reply:
Good point. We are having this conversation offline to try and get away from using envars here (and in the lines above to get rank et al) as the names are too specific to given implementations (and even versions of those implementations) and thus fragile.

Reply:
Yes, this is intended to be a temporary solution for 4.x only. A discussion is going on for how to work with 5.x:  https://github.com/open-mpi/ompi/pull/8359.